### PR TITLE
Create GIDAccountDetailType enumeration and its mapping.

### PR DESCRIPTION
**Pull request by brnnmrls**
The enumeration `GIDAccountDetailType` is created and mapped to scope "https://www.googleapis.com/auth/verified.age.over18.standard". GIDVerifiableAccountDetail takes care of three functions: 
1. `-[GIDVerifiableAccountDetail initWithAccountDetailType:]`
Designated initializer of the GIDVerifiableAccountDetail class
2. `+[GIDVerifiableAccountDetail scopeMapping:]`
Returns a dictionary mapping account detail type to scope.
3. `-[GIDVerifiableAccountDetail retrieveScope:]`
Returns scope of account detail needed to verify.

### CODE REVIEW

File: GoogleSignIn/Sources/GIDVerifyAccountDetail/Implementations/GIDVerifiableAccountDetail.m [hunk 1]
```diff
@@ -17,4 +17,25 @@
...
+
+- (instancetype)initWithAccountDetailType:(GIDAccountDetailType)accountDetailType {
+   self = [super init];
+   if (self) {
+       _accountDetailType = accountDetailType;
+   }
+   return self;
+}
+
+// Returns dictionary mapping account detail type to scope.
```
Comment (mdmathias) [inline GoogleSignIn/Sources/GIDVerifyAccountDetail/Implementations/GIDVerifiableAccountDetail.m:29]: I don't think this comment is necessary.

File: GoogleSignIn/Sources/GIDVerifyAccountDetail/Implementations/GIDVerifiableAccountDetail.m [hunk 2]
```diff
@@ -17,4 +17,25 @@
...
++ (NSDictionary<NSNumber *, NSString *> *)scopeMapping {
+  return @{
+    @(GIDAccountDetailTypeAgeOver18) : @"https://www.googleapis.com/auth/verified.age.over18.standard"
```
Comment (mdmathias) [inline GoogleSignIn/Sources/GIDVerifyAccountDetail/Implementations/GIDVerifiableAccountDetail.m:32]: `@"https://www.googleapis.com/auth/verified.age.over18.standard"` should be a string constant.

File: GoogleSignIn/Sources/GIDVerifyAccountDetail/Implementations/GIDVerifiableAccountDetail.m [hunk 3]
```diff
@@ -17,4 +17,25 @@
...
+  };
+}
+
+- (NSString *)retrieveScope {
```
Comment (mdmathias) [inline GoogleSignIn/Sources/GIDVerifyAccountDetail/Implementations/GIDVerifiableAccountDetail.m:36]: Alternative implementation to consider:

```
- (NSString *)scope {
  switch (self.accountDetailType) {
    case GIDAccountDetailTypeAgeOver18:
      return kGIDAccountDetailTypeAgeOver18Scope;
    default:
      return nil;
  }
}
```

I think this implementation is a little cleaner since we don't need an auxiliary method to store and create a dictionary every time we want the scope string -- IMHO the enumeration we have already does that.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifiableAccountDetail.h [hunk 4]
```diff
@@ -16,7 +16,29 @@
...
+/// Initializes a new GIDVerifiableAccountDetail object with the given
+/// account detail type.
+///
+/// @param accountDetailType The type of account detail that will be verified.
+- (instancetype)initWithAccountDetailType:(GIDAccountDetailType)accountDetailType;
+
+/// Retrieves the scope required to verify the account detail.
+///
+/// @return A string representing the scope required to verify the account detail.
+- (NSString *)retrieveScope;
```
Comment (mdmathias) [inline GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifiableAccountDetail.h:42]: `-[GIDVerifiableAccountDetail retrieveScope]` is essentially a getter for an "undeclared" property named `scope`. Therefore, I suggest renaming to `scope`.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifiableAccountDetail.h [hunk 5]
```diff
@@ -16,7 +16,29 @@
...
+    // Verifies the user is 18 years of age or older.
+    GIDAccountDetailTypeAgeOver18
+    // Potential future account details can be added here 
+};
+
+/// Helper object used to hold the enumeration representing a list of
 /// account details that Google can verify via GSI.
 @interface GIDVerifiableAccountDetail : NSObject
+
+// The type of account detail that will be verified.
```
Comment (mdmathias) [inline GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifiableAccountDetail.h:30]: Three forward slashes: `///`.

File: GoogleSignIn/Sources/GIDVerifyAccountDetail/Implementations/GIDVerifiableAccountDetail.m (hunk 1 above)
Comment (brnnmrls) [reply GoogleSignIn/Sources/GIDVerifyAccountDetail/Implementations/GIDVerifiableAccountDetail.m:29]: Done.

File: GoogleSignIn/Sources/GIDVerifyAccountDetail/Implementations/GIDVerifiableAccountDetail.m (hunk 2 above)
Comment (brnnmrls) [reply GoogleSignIn/Sources/GIDVerifyAccountDetail/Implementations/GIDVerifiableAccountDetail.m:32]: Done.

File: GoogleSignIn/Sources/GIDVerifyAccountDetail/Implementations/GIDVerifiableAccountDetail.m (hunk 3 above)
Comment (brnnmrls) [reply GoogleSignIn/Sources/GIDVerifyAccountDetail/Implementations/GIDVerifiableAccountDetail.m:36]: Ah, I see what you mean, also this is definitely simpler so I've swapped. Done.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifiableAccountDetail.h (hunk 4 above)
Comment (brnnmrls) [reply GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifiableAccountDetail.h:42]: Done.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifiableAccountDetail.h (hunk 5 above)
Comment (brnnmrls) [reply GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifiableAccountDetail.h:30]: Done.

File: GoogleSignIn/Sources/GIDVerifyAccountDetail/Implementations/GIDVerifiableAccountDetail.m [hunk 6]
```diff
@@ -16,5 +16,25 @@
 
 #import "GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifiableAccountDetail.h"
 
+static NSString *const kAgeVerificationScope = @"https://www.googleapis.com/auth/verified.age.over18.standard";
```
Comment (mdmathias) [inline GoogleSignIn/Sources/GIDVerifyAccountDetail/Implementations/GIDVerifiableAccountDetail.m:19]: Probably should update this to `kAccountDetailTypeAgeOver18Scope` since there may be other ages that are verifiable in the future.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifiableAccountDetail.h [hunk 7]
```diff
@@ -16,7 +16,29 @@
...
+/// Initializes a new GIDVerifiableAccountDetail object with the given
+/// account detail type.
+///
+/// @param accountDetailType The type of account detail that will be verified.
+- (instancetype)initWithAccountDetailType:(GIDAccountDetailType)accountDetailType;
+
+/// Retrieves the scope required to verify the account detail.
+///
+/// @return A string representing the scope required to verify the account detail.
+- (NSString *)scope;
```
Comment (mdmathias) [inline GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifiableAccountDetail.h:42]: Note that you return `nil` from the implementation but do not declare the return type as `nullable`: `- (nullable NSString *)scope;`. The implementation should also list the return type as `nullable`.

File: GoogleSignIn/Tests/Unit/GIDVerifiableAccountDetailTest.m [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+- (void)testDesignatedInitializer {
+  GIDVerifiableAccountDetail *detail = [[GIDVerifiableAccountDetail alloc] initWithAccountDetailType:GIDAccountDetailTypeAgeOver18];
+  XCTAssertNotNil(detail);
+  XCTAssertEqual(detail.accountDetailType, GIDAccountDetailTypeAgeOver18);
+}
+
+- (void)testScopeRetrieval {
+  GIDVerifiableAccountDetail *detail = [[GIDVerifiableAccountDetail alloc] initWithAccountDetailType:GIDAccountDetailTypeAgeOver18];
+  NSString *retrievedScope = [detail scope];
+  NSString *expectedScope = @"https://www.googleapis.com/auth/verified.age.over18.standard";
```
Comment (mdmathias) [inline GoogleSignIn/Tests/Unit/GIDVerifiableAccountDetailTest.m:19]: Can you use the string constant you made instead?

File: GoogleSignIn/Tests/Unit/GIDVerifiableAccountDetailTest.m [hunk 9]
```diff
@@ -0,0 +1,23 @@
+#import <XCTest/XCTest.h>
+
+#import "GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifiableAccountDetail.h"
+
+@interface GIDVerifiableAccountDetailTests : XCTestCase
+@end
+
+@implementation GIDVerifiableAccountDetailTests
```
Comment (mdmathias) [inline GoogleSignIn/Tests/Unit/GIDVerifiableAccountDetailTest.m:8]: I think it would be worth testing the edge case that the provided account detail yields a `nil` value when asking for the `scope`.

**Review by mdmathias — CHANGES_REQUESTED** [review]

File: GoogleSignIn/Sources/GIDVerifyAccountDetail/Implementations/GIDVerifiableAccountDetail.m (hunk 6 above)
Comment (brnnmrls) [reply GoogleSignIn/Sources/GIDVerifyAccountDetail/Implementations/GIDVerifiableAccountDetail.m:19]: Done.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifiableAccountDetail.h [hunk 10]
```diff
@@ -16,7 +16,29 @@
 
 #import <Foundation/Foundation.h>
 
-/// Helper object used to hold GIDAccountDetailType representing a list of
+/// An enumeration defining the types of account details Google can verify.
+typedef NS_ENUM(NSInteger, GIDAccountDetailType) {
+  // Verifies the user is 18 years of age or older.
```
[the remark this answers is not in the document]
Comment (brnnmrls) [reply GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifiableAccountDetail.h:21]: Done.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifiableAccountDetail.h [hunk 11]
```diff
@@ -16,7 +16,29 @@
...
+  GIDAccountDetailTypeAgeOver18
+  // Potential future account details can be added here
```
[the remark this answers is not in the document]
Comment (brnnmrls) [reply GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifiableAccountDetail.h:23]: Done.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifiableAccountDetail.h (hunk 7 above)
Comment (brnnmrls) [reply GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifiableAccountDetail.h:42]: Done.

File: GoogleSignIn/Tests/Unit/GIDVerifiableAccountDetailTest.m (hunk 8 above)
Comment (brnnmrls) [reply GoogleSignIn/Tests/Unit/GIDVerifiableAccountDetailTest.m:19]: Done.

File: GoogleSignIn/Tests/Unit/GIDVerifiableAccountDetailTest.m (hunk 9 above)
Comment (brnnmrls) [reply GoogleSignIn/Tests/Unit/GIDVerifiableAccountDetailTest.m:10]: Done.

**Review by mdmathias — CHANGES_REQUESTED** [review]

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifiableAccountDetail.h [hunk 12]
```diff
@@ -16,7 +16,34 @@
...
-/// Helper object used to hold GIDAccountDetailType representing a list of
+NS_ASSUME_NONNULL_BEGIN
+
+/// An enumeration defining the types of account details Google can verify.
+typedef NS_ENUM(NSInteger, GIDAccountDetailType) {
+  /// User account detail for age over 18.
+  GIDAccountDetailTypeAgeOver18,
+};
+
+extern NSString *const kAccountDetailTypeAgeOver18Scope;
```
Comment (mdmathias) [inline GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifiableAccountDetail.h:27]: Everything in a header file should have a doc comment: `/// String scope representing...`.

**Review by mdmathias — APPROVED** [review]
